### ISSUE TITLE
Weird postgres behaviour with NULL as default value

### ISSUE THREAD

**Issue by fabn** [ankane/strong_migrations#199]
Hi, we're using `strong_migrations` gem and it's very useful, so thanks for your work.

We were just bitten by a small downtime not detected by this gem, it was obviously our fault but it could be added to the gem to save other users.

We deployed this migration in production and it locked our system for about 40 seconds while running.

```ruby
add_column :products, :test_price, :decimal, precision: 16, scale: 4, default: nil
# This will obviously works
add_column :products, :test_price, :decimal, precision: 16, scale: 4
```

We debugged what happens under the hood in postgres (version 9.6) and we found out that these two queries have a different order of magnitude

```
D, [2022-09-21T18:13:32.275103 #49796] DEBUG -- :    (1575.6ms)  ALTER TABLE "products" ADD "test_price_cents" decimal(16,4) DEFAULT NULL
   -> 1.5773s
D, [2022-09-21T18:13:52.684874 #49848] DEBUG -- :    (1.0ms)  ALTER TABLE "products" ADD "test_price_cents" decimal(16,4)
   -> 0.0021s
```

Test were conducted on a development machine with about ~700k records and no other queries running.

So it seems that postgres will actually rewrite whole table even if default value is `NULL`.

I suggest to change the check in [this line](https://github.com/ankane/strong_migrations/blob/b978c08d9e68dd8de9042abb2c6e573822a9f740/lib/strong_migrations/checks.rb#L37) to something like `options.has_key?(:default)` instead of `default.nil?` to prevent this issue for future users.

Thanks.

**Comment by fabn** [comment]
BTW same query in postgres 14 leads to these times

```
D, [2022-09-21T18:31:31.603984 #51664] DEBUG -- :    (1.0ms)  ALTER TABLE "products" ADD "test_price_cents" decimal(16,4) DEFAULT NULL
   -> 0.0016s
D, [2022-09-21T18:30:10.932969 #51525] DEBUG -- :    (0.8ms)  ALTER TABLE "products" ADD "test_price_cents" decimal(16,4)
   -> 0.0015s
```

**Comment by ankane** [comment]
Hey @fabn, thanks for reporting! Just pushed a fix. This affects Postgres < 11. The current version of Strong Migrations only supports Postgres 10+, so you'll probably want to fork the gem and apply a similar fix if your application uses 9.6.
